src/Raven.Server/Documents/Sharding/ShardedContext.cs

```diff
@@ -183,10 +183,10 @@ public List<string> GetShardedDatabaseNames()
             return list;
         }
 
-        public async Task<string> GetLastDocumentChangeVectorForCollection(string collection)
+        public async Task<LastChangeVectorForCollectionCombinedResult> GetLastDocumentChangeVectorForCollection(string collection)
         {
-            var res = await ShardExecutor.ExecuteParallelForAllAsync(new ShardedLastChangeVectorForCollectionOperation(collection));
-            return res.LastChangeVector;
+            var res = await ShardExecutor.ExecuteParallelForAllAsync(new ShardedLastChangeVectorForCollectionOperation(collection, DatabaseName));
+            return res;
```

Comment:
Don't put this directly here.
Create file `ShardedContext.Subscriptions.cs` & `public partial class ShardedContext` and there create `ShardedSubscription class`. similar to `ShardedStreaming`